### PR TITLE
Error logger

### LINKED ISSUE #7: Add error handling + logging to the Query Facade
**What you gotta do:**
- add a logging framework (Zoe + Sen selected [Winston](https://github.com/winstonjs/winston)) 
       - let the logging framework write to a file
  
- add in error handling for the QueryFacade
       - log errors
       - set up a retry framework - (frameworks to consider: axios retry, a decorator that handles retries)
       - do not retry if the status code is 401, 404 or 405 - these do not succeed even if they are retried

### PR DESCRIPTION
**Pull request by z-arnott**
This PR should be reviewed after PR #10 
Closes #7 

Uses Winston for logging.
The logger is configured to save logs a file in /logs and print logs to the console.
Each log includes severity level (ie. error, info, warn etc), timestamp, and message.
The logger can be called by importing {logger} module.

### CODE REVIEW

File: src/utils/logger.utils.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+const alignedWithColorsAndTime = winston.format.combine(
+  colorize(),
+  timestamp(),
+  align(),
+  printf((info) => {
+    const {
+      timestamp, level, message, ...args
+    } = info;
+
+    const ts = timestamp.slice(0, 19).replace('T', ' ');
```
Comment (rhass-uta) [inline src/utils/logger.utils.ts:21]: explain what you're slicing and why 0-19 9 (assuming you're slicing time in 20s?)

Comment (z-arnott) [reply src/utils/logger.utils.ts:21]: it timestamps the log in printable format. this is the out of the box timestamp implementation from winston examples

File: src/utils/logger.utils.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+          }),
+          prettyPrint()
+        )
+      }),
+      new winston.transports.Console({
+        format:  combine(
+          timestamp({
+            format: "MMM-DD-YYYY HH:mm:ss",
+          }), 
+          alignedWithColorsAndTime
```
Comment (rhass-uta) [inline src/utils/logger.utils.ts:46]: maybe im confused - but what's the actual different between the output for prettyPrint and the output for alignedWithColorsAndTime

Comment (z-arnott) [reply src/utils/logger.utils.ts:46]: I'll add some comments here in the code
but these are two different transports
winston.transports.File makes a log that saves to file (JSON) and the file needs to be formatted with prettyprint

winston.transports.Console prints to console which needs string formatting with alighedWithConolorsAndTime

**Review by rhass-uta — APPROVED** [review]
Looking good, just give a little bit more context to the different functions in the logger.utils.ts but otherwise it's written pretty clearly and simply

**Comment by z-arnott** [comment]
> Looking good, just give a little bit more context to the different functions in the logger.utils.ts but otherwise it's written pretty clearly and simply

thanks I added some comments for clarity!

**Pull request merged**